exchange/src/exchange/wrappers/handle/wrapper_handle.cpp

```diff
@@ -68,8 +78,16 @@ WrapperHandle::block_on_init()
     throw std::runtime_error("Received non-init message on initialization");
 }
 
-WrapperHandle::WrapperHandle(const std::vector<std::string>& args)
+WrapperHandle::WrapperHandle(
+    const std::vector<std::string>& args, const std::optional<std::string> optional_algo
```

Comment:
nit: it's already an optional, no reason to say optional again. Also, if you're passing by value, don't use const. Also, you have an optional in a constructor by overloading, not by making it an optional